pkg/controller/stack/stack_controller.go

```diff
@@ -1085,6 +1086,75 @@ func (sess *reconcileStackSession) SetEnvRefsForWorkspace(ctx context.Context, w
 	return nil
 }
 
+func (sess *reconcileStackSession) resolveConfigRefs(ctx context.Context) ([]ConfigKeyValue, error) {
+	allConfigs := make([]ConfigKeyValue, 0)
+	for k, ref := range sess.stack.ConfigRefs {
+		// ConfigMap and Structured are special config cases, so they are checked first
+		switch ref.SelectorType {
+		case shared.ConfigResourceSelectorConfigMap:
+			configMapRef := ref.ConfigMapRef
+			if configMapRef != nil {
+				var config corev1.ConfigMap
+				if err := sess.kubeClient.Get(ctx, types.NamespacedName{Name: configMapRef.Name, Namespace: configMapRef.Namespace}, &config); err != nil {
+					return nil, fmt.Errorf("Failed to get the ConfigMap %s on namespace %s: %w", configMapRef.Name, configMapRef.Namespace, err)
+				}
+				// assumes the whole configmaps's data is the config content; try to read as a conventional stack yaml config
+				var configMapContent map[string]any
+				if err := yaml.Unmarshal([]byte(config.Data[configMapRef.Key]), &configMapContent); err != nil {
+					return nil, fmt.Errorf("Failed to read the ConfigMap content as a stack YAML config. Namespace=%s Name=%s: %w", configMapRef.Namespace, configMapRef.Name, err)
+				}
+				structuredConfig := StructuredConfig(configMapContent).Flatten()
+				allConfigs = append(allConfigs, structuredConfig...)
+			}
+		case shared.ConfigResourceSelectorStructured:
```

Comment:
I'm a little confused about the need for both "structured" and "literal".

The legacy `resolveResourceRef` logic doesn't do any special handling for "literal", but we have an opportunity to be more flexible here. 

In other words, it's odd to have two ways to express the same thing:
```
configRefs:
   stack-config:
     type: Literal
     value: foo
--
configRefs:
   stack-config:
     type: Structured
     value: foo
```

But this is net-new:
```
configRefs:
   stack-config:
     type: Structured
     value:
        my-config:
          my-nested-field:
             field: "value"
```

It would seem reasonable IMO to basically remove structured and just change how we handle literals (basically replacing this block with  `case ResourceSelectorLiteral`).

Reply:
I thought about this option first but the reason why I've preferred to introduce the "structured" type was don't change the "literal" behavior, because it's already used by `EnvRefs` and `SecretRefs` fields, and for these use cases the "literal" option looks good enough (specially for env var case)

For these use cases just passing a literal, simple string value makes sense, but a structured, complex value does not, I think...so I didn't want to change that, and introduced the new "structured" option (I used this word because Pulumi docs also use it to [reference about complex configs](https://www.pulumi.com/docs/concepts/config/#structured-configuration). So the main reason was backward compatibility.

but you're right, it sounds a bit strange. maybe we could handle "literal" for configs as you suggested and also keep the legacy "literal" behavior for the other `Refs` fields. WDYT?